lib/workers/package-file/dep-type.js

```diff
@@ -104,6 +106,9 @@ function getDepConfig(depTypeConfig, dep) {
           }
         }
       }
+      if (applyRule !== false && matchVersion) {
+        applyRule = matchesSemver(dep.currentVersion, matchVersion);
```

Comment:
Let's test how this works with following combinations:
- currentVersion is range that overlaps with matchVersion range
- currentVersion is pinned version that overlaps with matchVersion range
- currentVersion and matchVersion are the same pinned version (e.g. v1.2.3)